Connectors/Revit/Speckle.Connectors.RevitShared/Operations/Send/RevitRootObjectBuilder.cs

```diff
@@ -56,91 +59,123 @@ CancellationToken cancellationToken
+          currentSettings with
+          {
+            ReferencePointTransform = atomicObjectByDocumentAndTransform.Transform,
+            Document = atomicObjectByDocumentAndTransform.Doc
+          }
+        )
+      )
       {
-        if (!SupportedCategoriesUtils.IsSupportedCategory(revitElement.Category))
-        {
-          var cat = revitElement.Category != null ? revitElement.Category.Name : "No category";
-          results.Add(
-            new(
-              Status.WARNING,
-              revitElement.UniqueId,
-              cat,
-              null,
-              new SpeckleException($"Category {cat} is not supported.")
-            )
-          );
-          skippedObjectCount++;
-          continue;
-        }
-
-        Base converted;
-        if (sendConversionCache.TryGetValue(sendInfo.ProjectId, applicationId, out ObjectReference? value))
-        {
-          converted = value;
-          cacheHitCount++;
-        }
-        else
+        var atomicObjects = atomicObjectByDocumentAndTransform.Elements;
+        foreach (Element revitElement in atomicObjects)
         {
-          converted = converter.Convert(revitElement);
-          converted.applicationId = applicationId;
+          cancellationToken.ThrowIfCancellationRequested();
+          string applicationId = revitElement.UniqueId;
+          string sourceType = revitElement.GetType().Name;
+          try
+          {
+            if (!SupportedCategoriesUtils.IsSupportedCategory(revitElement.Category))
+            {
+              var cat = revitElement.Category != null ? revitElement.Category.Name : "No category";
+              results.Add(
+                new(
+                  Status.WARNING,
+                  revitElement.UniqueId,
+                  cat,
+                  null,
+                  new SpeckleException($"Category {cat} is not supported.")
+                )
+              );
+              skippedObjectCount++;
+              continue;
+            }
+
+            Base converted;
+            if (sendConversionCache.TryGetValue(sendInfo.ProjectId, applicationId, out ObjectReference? value))
+            {
+              converted = value;
+              cacheHitCount++;
+            }
+            else
+            {
+              converted = converter.Convert(revitElement);
+              converted.applicationId = applicationId;
+            }
+
+            var collection = sendCollectionManager.GetAndCreateObjectHostCollection(revitElement, rootObject);
+
+            collection.elements.Add(converted);
+            results.Add(new(Status.SUCCESS, applicationId, sourceType, converted));
+          }
+          catch (Exception ex) when (!ex.IsFatal())
+          {
+            logger.LogSendConversionError(ex, sourceType);
+            results.Add(new(Status.ERROR, applicationId, sourceType, null, ex));
+          }
+
+          onOperationProgressed.Report(new("Converting", (double)++countProgress / atomicObjects.Count));
```

Comment:
We will probably need to update to account for total elements across all documents - if possible. We will see